var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py

```diff
@@ -13,21 +11,28 @@ class PyPennylaneLightningKokkos(CMakePackage, PythonExtension, CudaPackage, ROC
 
     homepage = "https://docs.pennylane.ai/projects/lightning-kokkos"
     git = "https://github.com/PennyLaneAI/pennylane-lightning-kokkos.git"
-    url = "https://github.com/PennyLaneAI/pennylane-lightning-kokkos/archive/refs/tags/v0.32.0.tar.gz"
-
+    urls = [
+        "https://github.com/PennyLaneAI/pennylane-lightning-kokkos/archive/refs/tags/v0.32.0.tar.gz"
+        "https://github.com/PennyLaneAI/pennylane-lightning/archive/refs/tags/v0.35.1.tar.gz"
+    ]
```

Comment:
There are basically three options for specifying urls.  

First is with the `url` property (as you had before).  Spack automatically substitutes version names (when it can).  

Second is to add a `url` argument to the `version` directive.  You can have both the `url` property (with spack version substitution as the default) and use the argument to override it on a directive basis if that makes sense.

The third option is to implement a `url_for_version` method.

You can see more on the options in the docs (https://spack.readthedocs.io/en/latest/packaging_guide.html#version-urls).

Reply:
FWIW. I can confirm all of the version sha256s when I remove the `urls` property and add (at around line 101) the following:

```
    def url_for_version(self, version):
        extra = "-kokkos" if version <= Version("0.32.0") else ""
        return f"https://github.com/PennyLaneAI/pennylane-lightning{extra}/archive/refs/tags/v{version}.tar.gz"
```

Reply:
Nice suggestion, thanks @tldahlgren . I didn't know about this option. 

I was wondering, 
- is there a one-liner function in `spack` to format the code (i.e. running `black`, `isort`, etc.)? 
- is there a quick way to validate all `sha256` (e.g. `spack checksums <package>`)?

Reply:
> * is there a one-liner function in `spack` to format the code (i.e. running `black`, `isort`, etc.)?

You can try `spack style` (https://spack.readthedocs.io/en/latest/command_index.html#spack-style).  I haven't used it to change code so I count vouch for that but use a local `black` instead.  Isn't configured exactly the same so YMMV.


> * is there a quick way to validate all `sha256` (e.g. `spack checksums <package>`)?

You can try `spack checksum` (https://spack.readthedocs.io/en/latest/command_index.html#spack-checksum).  I have a more complicated script (WIP) I use for additional checks that leverages `spack checksum`.

Reply:
So I rebuilt all version and caught a compatibility issue for 0.31 ;) Everything should work fine now.

Reply:
Found few more version bounds issues running tests. All good now.